middleware/sunset.go

```diff
@@ -0,0 +1,29 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+)
+
+// Sunset set Deprecation/Sunset header to response
+// This can be used to enable Sunset in a route or a route group
+// For more at: https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-deprecation-header-02
+func Sunset(sunsetAt *time.Time, links ...string) func(http.Handler) http.Handler {
```

Comment:
does this have to be a pointer?

Reply:
Yes, pointer was not necessary. Only time.Time is enough for this usecase 👍🏻